data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -70,31 +109,59 @@ void viewlist()  // function to display values
     }
 }
 //////////////////////////////////////////////////////////////////////
-
+int test(){
+    insert(1,1);
+    insert(2,2);
+    deletion(2);
+    viewlist;
+    return 0;
+}
 int main()
 {
-    int n;
-    while (1)
+    int n=0,pos=0,p=0,num=0,c=0;
+    printf("\n1.self test mode");
+    printf("\n2.interactive mode");
+    printf("\nenter your choice:");
+    scanf("%d",&c);
+    if(c==1)
+    {
+       assert(test()==0);
```

Comment:
All the `assert` calls should be in the `test` function, which should return `void`.
In the `main` function, all you have to do is call the function if the self-test mode was chosen.